pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -803,6 +816,31 @@ func uploadCRNameForSnapshotPEID(snapshotPEID astrolabe.ProtectedEntityID) (stri
 	return "upload-" + snapshotPEID.GetSnapshotID().String(), nil
 }
 
+func decodePeIdFromSnapshotID(snapshotID astrolabe.ProtectedEntitySnapshotID, logger logrus.FieldLogger) (astrolabe.ProtectedEntityID, error) {
+	var peId astrolabe.ProtectedEntityID
+	snapshotID64Str := snapshotID.String()
+	snapshotIDBytes, err := base64.RawStdEncoding.DecodeString(snapshotID64Str)
+	if err != nil {
+		errorMsg := fmt.Sprintf("decodePeIdFromSnapshotID: Could not decode snapshot ID encoded string %s", snapshotID64Str)
+		logger.WithError(err).Error(errorMsg)
+		return peId, errors.Wrap(err, errorMsg)
+	}
+	peId, err = astrolabe.NewProtectedEntityIDFromString(string(snapshotIDBytes))
+	if err != nil {
+		errorMsg := fmt.Sprintf("decodePeIdFromSnapshotID: Could not translate decoded snapshotID %s into pe-id", string(snapshotIDBytes))
+		logger.WithError(err).Error(errorMsg)
+		return peId, errors.Wrap(err, errorMsg)
+	}
+	logger.Infof("decodePeIdFromSnapshotID: Successfully translated snapshotID %s into pe-id: %s", snapshotID.String(), peId.String())
+	if peId.HasSnapshot() && peId.GetPeType() != "ivd" {
+		// This is exclusively called only in supervisor and vanilla execution path.
+		// This is done to allow deletion of for example a guest cluster backup in a vanilla setup.
```

Comment:
How did you determine this is supervisor/vanilla and it is deletion guest backup in vanilla?

Reply:
It's based on the pe-Id format. `decodePeIdFromSnapshotID` is exclusively called when we are in Vanilla/Supervisor. If the first decode of pe-id leads to "ivd" type that means we are deleting the vanilla backup from vanilla/sup setup or supervisor backup from vanilla/sup setup, if not, the first decode leads to "paravirt-pv" type pe-id, decoding the paravirt-pv pe-id leads to the actual ivd we need to delete.

example
Incoming snapshotID for Guest created backup: cGFyYXZpcnQtcHY6cHZjLTQwNTY1NmNjLThkZjgtNDE0YS1hYzU3LWU3NWMwOGQ5MWQ0NjphWFprT2pjek1ESmhNV1pqTFdGa05qRXRORGswTnkwNFlUQTFMV1V3TTJVellUZGxZV1V4TlRvNU5ERmhOV00yWVMwNE5qUTFMVFExT1RNdE9HTXdZUzA1WTJZd1pXSmhOVEZtTXpZ

Decodes to:
paravirt-pv:pvc-405656cc-8df8-414a-ac57-e75c08d91d46:aXZkOjczMDJhMWZjLWFkNjEtNDk0Ny04YTA1LWUwM2UzYTdlYWUxNTo5NDFhNWM2YS04NjQ1LTQ1OTMtOGMwYS05Y2YwZWJhNTFmMzY 

Incoming snapshotID for Vanilla/Sup Backup: aXZkOjczMDJhMWZjLWFkNjEtNDk0Ny04YTA1LWUwM2UzYTdlYWUxNTo5NDFhNWM2YS04NjQ1LTQ1OTMtOGMwYS05Y2YwZWJhNTFmMzY

decodes to:
ivd:7302a1fc-ad61-4947-8a05-e03e3a7eae15:941a5c6a-8645-4593-8c0a-9cf0eba51f36